internal/resources/azure/sql_managed_instance.go

```diff
@@ -0,0 +1,196 @@
+// It uses the `infracost_usage` struct tags to populate data into the SQLManagedInstance.
+func (r *SQLManagedInstance) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+// BuildResource builds a schema.Resource from a valid SQLManagedInstance struct.
+// This method is called after the resource is initialised by an IaC provider.
+// See providers folder for more information.
+func (r *SQLManagedInstance) BuildResource() *schema.Resource {
+	costComponents := r.costComponents()
+
+	return &schema.Resource{
+		Name: r.Address,
+		UsageSchema: []*schema.UsageItem{
+			{Key: "backup_storage_gb", DefaultValue: 0, ValueType: schema.Int64},
+			{Key: "long_term_retention_storage_gb", DefaultValue: 0, ValueType: schema.Int64},
+		},
+		CostComponents: costComponents,
+	}
+}
+
+func (r *SQLManagedInstance) costComponents() []*schema.CostComponent {
+
+	costComponents := []*schema.CostComponent{
+		{
+			Name:           fmt.Sprintf("Compute (%s %s Cores)", strings.ToTitle(r.SKU), strconv.FormatInt(*r.Cores, 10)),
+			Unit:           "hours",
+			UnitMultiplier: decimal.NewFromInt(1),
+			HourlyQuantity: decimalPtr(decimal.NewFromInt(1)),
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr(vendorName),
+				Region:        strPtr(r.Region),
+				Service:       strPtr(sqlMIServiceName),
+				ProductFamily: strPtr(sqlMIProductFamily),
+				AttributeFilters: ([]*schema.AttributeFilter{
+					{Key: "productName", Value: r.productDescription()},
+					{Key: "skuName", Value: r.meteredName()},
+				}),
+			},
+			PriceFilter: priceFilterConsumption,
+		},
+	}
+
+	if r.BackupStorageGb != nil {
+		costComponents = append(costComponents, r.sqlMIStorageCostComponent(), r.sqlMIBackupCostComponent())
+	}
+
+	if r.LicenceType == "LicenseIncluded" {
+		costComponents = append(costComponents, r.sqlMILicenseCostComponent())
+	}
+	if r.LongTermRetentionStorageGB != nil {
+		costComponents = append(costComponents, r.sqlMILongTermRetentionStorageGBCostComponent())
+	}
+	return costComponents
+}
+
+func (r *SQLManagedInstance) productDescription() *string {
+
+	productDescription := ""
+	if strings.Contains(r.SKU, "GP") {
+		productDescription = "SQL Managed Instance General Purpose"
+	} else if strings.Contains(r.SKU, "BC") {
+		productDescription = "SQL Managed Instance Business Critical"
+	}
+
+	if strings.Contains(r.SKU, "Gen5") {
+		productDescription = fmt.Sprintf("%s - %s", productDescription, "Compute Gen5")
+	}
+	return strPtr(productDescription)
+}
+
+func (r *SQLManagedInstance) meteredName() *string {
+	meterName := fmt.Sprintf("%s %s", strconv.FormatInt(*r.Cores, 10), "vCore")
+	return strPtr(meterName)
+}
+
+func (r *SQLManagedInstance) sqlMIStorageCostComponent() *schema.CostComponent {
+
+	return &schema.CostComponent{
+		Name:            fmt.Sprintf("Storage %s Gb (first 32 Gb include)", strings.ToTitle(strconv.FormatInt(*r.StorageSizeInGb, 10))),
+		Unit:            "Unit of 32Gb",
```

Comment:
This was confusing to me as the output reads:

```
 ├─ Storage 64 Gb (first 32 Gb include)                              32  Unit of 32Gb         $4.38 
```
 
 I would suggest changing this to just `GB`
 
 ```
 ├─ Storage                          64  GB         $4.38 
```

and omitting the `first 32GB` text